### PR TITLE
PermissionManager

### LINKED ISSUE #111: Permission Manager
### Bug Report
Create Permission Manager to give developers information about what permissions are permitted in the specific HMI level.

### PR DESCRIPTION
**Pull request by nickschwab**
Fixes #111 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
Test each public method created and its integration with SdlManager

### Summary
PermissionManager implemented

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform) and this pull request adheres to the [Contributing Guide](./CONTRIBUTING.md).

### CODE REVIEW

File: lib/js/src/manager/permission/PermissionManagerBase.js [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+            this._checkState();
+            this._notifyListeners(this._currentPermissionItems, previousHmiLevel, this._currentPermissionItems, this._currentHmiLevel);
+        };
+        this._lifecycleManager.addRpcListener(FunctionID.OnHMIStatus, this._onHMIStatusListener);
+
+        this._onPermissionsChangeListener = (notification) => {
+            const permissionItems = notification.getPermissionItem();
+            const previousPermissionItems = JSON.parse(JSON.stringify(this._currentPermissionItems));
+            const requireEncryptionAppLevel = notification.getRequireEncryption();
+            this._encryptionRequiredRpcs.splice(0, this._encryptionRequiredRpcs.length);
```
Comment (crokita) [inline lib/js/src/manager/permission/PermissionManagerBase.js]: maybe state that this line clears the array out?

File: lib/js/src/util/Uuid.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/* eslint-disable */
```
[the remark this answers is not in the document]
Comment (nickschwab) [reply lib/js/src/util/Uuid.js]: Resolved in https://github.com/smartdevicelink/sdl_javascript_suite/pull/116/commits/a5f638b42612168418303acc8d49b14740469b00

File: examples/node/hello-sdl/AppClient.js [hunk 3]
```diff
@@ -128,6 +149,14 @@ class AppClient {
             setTimeout(resolve, timeout);
         });
     }
+
+    _logPermissions () {
```
[the remark this answers is not in the document]
Comment (nickschwab) [reply examples/node/hello-sdl/AppClient.js]: Resolved in https://github.com/smartdevicelink/sdl_javascript_suite/pull/116/commits/a5f638b42612168418303acc8d49b14740469b00

File: lib/js/src/manager/permission/PermissionManagerBase.js (hunk 1 above)
Comment (nickschwab) [reply lib/js/src/manager/permission/PermissionManagerBase.js]: Resolved in https://github.com/smartdevicelink/sdl_javascript_suite/pull/116/commits/a5f638b42612168418303acc8d49b14740469b00

**Pull request merged**